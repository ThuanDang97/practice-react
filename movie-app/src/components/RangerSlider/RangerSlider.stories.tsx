import { Story } from '@storybook/react';
import RangerSlider from './RangerSlider';
import { TProps } from './RangerSlider';

export default {
  title: 'Components/RangerSlider',
  component: RangerSlider,
};

const Template: Story<TProps> = (args): JSX.Element => <RangerSlider {...args} />;

export const Basic: Story<TProps> = Template.bind({});
Basic.args = {
  label: 'Score',
  min: 0,
  max: 10,
  defaultValue: 10,
  step: 1,
};
