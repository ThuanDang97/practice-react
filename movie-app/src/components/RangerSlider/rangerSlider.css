.ranger-slider {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.range {
  width: 100%;
}

.range-value {
  display: block;
}
