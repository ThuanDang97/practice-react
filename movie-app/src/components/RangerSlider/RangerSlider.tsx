import { ChangeEvent, useState } from 'react';
import './rangerSlider.css';

export type TProps = {
  label: string;
  min: number;
  max: number;
  defaultValue: number;
  step: number;
  onHandleFilterUserScore: (e: ChangeEvent<HTMLInputElement>) => void;
};

const RangerSlider = (props: TProps): JSX.Element => {
  const { onHandleFilterUserScore, label, min, max, defaultValue, step } = props;
  const [value, setValue] = useState<number>(10);

  const onHandleChange = (e: ChangeEvent<HTMLInputElement>) => {
    onHandleFilterUserScore(e);
    setValue(Number(e.target.value));
  };

  return (
    <div className="ranger-slider">
      {label && <h3>{label}</h3>}
      <input
        type="range"
        className="range"
        min={min}
        max={max}
        step={step}
        defaultValue={defaultValue}
        onChange={onHandleChange}
      />
      <div className="range-value">{value}</div>
    </div>
  );
};
export default RangerSlider;
