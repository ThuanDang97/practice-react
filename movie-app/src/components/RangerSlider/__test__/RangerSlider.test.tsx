import { fireEvent, screen, render } from '@testing-library/react';
import { setup } from '../../../support/Setup';
import RangerSlider, { TProps } from '../RangerSlider';

describe('Test Ranger Slider component', () => {
  const defaultProps: TProps = {
    label: 'Title',
    min: 0,
    max: 10,
    defaultValue: 10,
    step: 1,
    onHandleFilterUserScore: jest.fn(),
  };

  test('Component Ranger Slider matches DOM Snapshot', () => {
    const component = <RangerSlider {...defaultProps} />;

    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  test('Component Ranger Slider should render correctly', () => {
    setup(<RangerSlider {...defaultProps} />);

    expect(screen.getByRole('heading')).toHaveTextContent('Title');
  });

  test('Component Ranger Slider should have default value is 10', () => {
    setup(<RangerSlider {...defaultProps} />);

    expect(screen.getByRole('slider')).toHaveValue(String(10));
  });

  test('Component Ranger Slider should called handleChange', async () => {
    // <input type = 'range' /> is not used by userEvent support to handle onChange,
    // so we will temporarily use fireEvent to handle onchange at this time
    // git issue: https://github.com/testing-library/user-event/issues/871
    setup(<RangerSlider {...defaultProps} />);

    fireEvent.change(screen.getByRole('slider'), { target: { value: 9 } });

    expect(defaultProps.onHandleFilterUserScore).toBeCalled();
    expect(screen.getByRole('slider')).toHaveValue(String(9));
  });
});
