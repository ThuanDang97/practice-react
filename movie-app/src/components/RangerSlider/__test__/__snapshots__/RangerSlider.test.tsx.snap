// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Ranger Slider component Component Ranger Slider matches DOM Snapshot 1`] = `
<div>
  <div
    class="ranger-slider"
  >
    <h3>
      Title
    </h3>
    <input
      class="range"
      max="10"
      min="0"
      step="1"
      type="range"
      value="10"
    />
    <div
      class="range-value"
    >
      10
    </div>
  </div>
</div>
`;
